cardano-node/cardano-node.cabal

```diff
@@ -133,6 +133,12 @@ library
                        -Wpartial-fields
                        -Wcompat
 
+  -- Pattern match checking has improved since 8.6.5. GHC now considers more
+  -- pattern matches as redundant than before, so disable this warning for older
+  -- versions of GHC that still consider them incomplete.
+  if impl(ghc < 8.10)
+    ghc-options:       -Wno-incomplete-patterns
+
```

Comment:
This doesn't work the way it should because for some reasons the warning flags are duplicated/overridden in `cabal.project`.

https://github.com/input-output-hk/cardano-node/blob/8d110471462f296f66bc91735426047a4cb1909a/cabal.project#L11-L24

I guess the advantage of putting them there is that it's easy to change all of them at once. However, I'm not sure whether that's the right thing to do: now the flags are in two places (`cabal.project` and the separate `.cabal` files). People wanting to change the flags in the `.`cabal` files (like me) will be surprised when their flags have no effect (as they're overridden by the flags in the `cabal.project` file).

I propose the following:
* We make sure all the warnings listed in `cabal.project` are enabled for each package component in this project.
* In `cabal.project`, we trim `ghc-options` to `-Werror`. All the warnings are already in the `.cabal` files, but `-Werror` is a switch that you don't want to have in each `.cabal` file.

I'll add a commit with this change. We can of course discuss this.

Reply:
Oh, I just discovered that the `.cabal` files include more warnings than the `cabal.project`, e.g., `-Wcompat`. So this duplication was already out of sync/date.

Reply:
See https://github.com/input-output-hk/cardano-node/pull/1990/commits/49699c5422e970de776e462f33915318046a3f67